packages/gatsby-theme-store/src/sdk/orderForm/useRemoveItemAnalytics.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback } from 'react'
+
+import { sendPixelEvent } from '../pixel/usePixelSendEvent'
+import type { OrderFormItemToAnalytics } from './useUpdateQuantityAnalytics'
+import { orderFormItemToPixelItem } from './useUpdateQuantityAnalytics'
+
+export interface RemoveItemWithAnalyticsParams<K, S> {
+  removeItem: (item: K) => S
+}
+
+export function removeItemWithAnalytics<
+  T extends OrderFormItemToAnalytics,
+  K,
+  S
+>({ removeItem }: RemoveItemWithAnalyticsParams<K, S>) {
+  return (item: T & K): S => {
+    const removeItemResult = removeItem(item)
+
+    const pixelEventItem = orderFormItemToPixelItem(item)
+
+    sendPixelEvent({
+      type: 'vtex:removeFromCart',
+      data: {
+        items: [pixelEventItem],
+      },
+    })
+
+    return removeItemResult
+  }
+}
+
+export function useRemoveItemAnalytics<
+  T extends OrderFormItemToAnalytics,
+  K,
+  S
+>({ removeItem }: RemoveItemWithAnalyticsParams<K, S>) {
+  const removeItemWithAnalyticsCallback = useCallback(
+    (item: T & K) => removeItemWithAnalytics({ removeItem })(item),
```

Comment:
Sorry, I didn't understand the necessity of abstraction of the `removeItemWithAnalytics` func. What's the advantage of curry this function instead of a not curried one?

```suggestion
    (item: T & K): S => {
    const removeItemResult = removeItem(item)

    const pixelEventItem = orderFormItemToPixelItem(item)

    sendPixelEvent({
      type: 'vtex:removeFromCart',
      data: {
        items: [pixelEventItem],
      },
    })

    return removeItemResult
  }
```

Reply:
If you agree, remove the `removeItemWithAnalytics` function

Reply:
The abstraction is there so if users want to use this helper function inside another hook or outside a react component, they can.